pallets/loans-ref/src/lib.rs

```diff
@@ -523,6 +565,33 @@ mod pallet {
 			Ok(())
 		}
 
+		#[pallet::weight(10_000)]
+		#[transactional]
+		pub fn modify(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			change_id: T::ChangeId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			T::ChangeGuard::released(pool_id, change_id)?;
+
+			let modification = LoanModifications::<T>::take(pool_id, change_id)
+				.ok_or(DispatchError::Other("Expected a stored modification"))?;
```

Comment:
I  would use a proper error at this level

Reply:
I did it in this way because, logically, this error can never happen and, therefore, can not be tested (not belong to the model). The `released` method will fail if the pair `pool_id`, `change_id` does not exist, so if you reach `LoanModifications::take()`, it always contains something. We could use an `unwrap()` here too but it's not "safe".

I tried to rationale this more deeply in our (maybe "my" at this moment) [rust guidelines](https://centrifuge.hackmd.io/sLMxV5WoTy2q57fJTfbNLQ?view#Proposal)